e3nn/util/datatypes.py

```diff
@@ -1,13 +1,19 @@
 from typing import NamedTuple, Optional
-
+from enum import Enum, auto
 
 class Chunk(NamedTuple):
     mul: int
     dim: int
     slice: Optional[slice] = None
 
+class TensorProductMode(Enum):
+    UUU = "uuu"
+    UVUV = "uvuv"
+    UVW = "uvw"
+    UVU_V = "uvu<v"
 
 class Path(NamedTuple):
     input_1_slice: Chunk
-    input_2_slice: Chunk
-    output_slice: Chunk
+    input_2_slice: Optional[Chunk] = None
+    output_slice: Optional[Chunk] = None
+    tensor_product_mode: TensorProductMode = TensorProductMode.UUU
```

Comment:
Can we set UVW as the default since thats what folks tend to use out of the box ?